test/unit/test_tpm2_policy.c

```diff
@@ -0,0 +1,374 @@
+//**********************************************************************;
+// Copyright (c) 2017, Intel Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+
+#include <errno.h>
+#include <stdarg.h>
+#include <stddef.h>
+
+#include <stdio.h>
+
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include <sapi/tpm20.h>
+
+#include "pcr.h"
+#include "tpm2_alg_util.h"
+#include "tpm2_policy.h"
+#include "tpm2_session.h"
+#include "tpm2_util.h"
+
+typedef struct pcr_data pcr_data;
+struct pcr_data {
+    TPML_DIGEST values;
+};
+
+typedef struct test_file test_file;
+struct test_file {
+    char *path;
+    FILE *file;
+};
+
+/* Passing tests and static data are hardcoded around this sel spec */
+#define PCR_SEL_SPEC "sha256:0,1,2,3"
+
+/*
+ * Dummy value for the session handle read by teh wrapped version of:
+ *   Tss2_Sys_StartAuthSession
+ */
+#define SESSION_HANDLE 0xDEADBEEF
+
+/* dummy handle for sapi context */
+#define SAPI_CONTEXT   ((TSS2_SYS_CONTEXT *)0xDEADBEEF)
+
+/* Any PCR read returns this value */
+static TPM2B_DIGEST pcr_value = { .size = 32, .buffer = { 0x96, 0xa7, 0xfa,
+        0xaf, 0x16, 0x09, 0xb6, 0x50, 0xa4, 0xf2, 0x88, 0xc0, 0x90, 0x4f, 0x04,
+        0x83, 0x6e, 0xca, 0xda, 0x2f, 0x49, 0x78, 0x06, 0x94, 0x86, 0xa2, 0xbb,
+        0x02, 0xf2, 0xf0, 0x43, 0xea } };
```

Comment:
Again, I think something like the following makes data structure declaration easier to read:

```
static TPM2B_DIGEST pcr_value = {
    .size = 32,
    .buffer = {
        0x96, 0xa7, 0xfa, 0xaf, 0x16, 0x09, 0xb6, 0x50, 0xa4, 0xf2,
        0x88, 0xc0, 0x90, 0x4f, 0x04, 0x83, 0x6e, 0xca, 0xda, 0x2f,
        0x49, 0x78, 0x06, 0x94, 0x86, 0xa2, 0xbb, 0x02, 0xf2, 0xf0,
        0x43, 0xea
     }
};
```